### PR TITLE
chore: Remove old examples dir

### LINKED ISSUE #3446: rename `examples` directory to more appropriate name
When new users arrive they often look for tutorials, readmes, samples, and examples.

We have an `examples` directory that came from the merge of the `cloak` codebase into `main`.
It contains examples of extensions.
We're not emphasizing extensions right now as we perfect the messaging and DX.
So we should rename this directory for now so new users are not confused into thinking these are relevant examples for Go SDK etc.

I propose renaming the directory to 1. `experimental` or  2. `extensions`. Or nested as 3. `experimental/extensions`.
Since we want to talk about extensions later on, I'd think option 1 or 3 is best.

@shykes @mircubed

### PR DESCRIPTION
**Pull request by jpadams**
fixes #3446 
Signed-off-by: Jeremy Adams <jeremy@dagger.io>

### CODE REVIEW

**Review by kpenfound — APPROVED** [review]
🔥

**Comment by aluzzardi** [comment]
@sipsma do we want to move those somewhere else?

**Review by sipsma — APPROVED** [review]
LGTM

@aluzzardi I think this is fine for now as we still have coverage of the project code paths in `project_test.go`. There's no need to maintain the examples for now. They'll always be in our git history if we want them back.

File: examples/nodejs/say.ts [hunk 1]
```diff
@@ -1,13 +0,0 @@
-import Client, {connect} from "@dagger.io/dagger"
```
Comment (sipsma) [inline examples/nodejs/say.ts:1]: Just noticed this one isn't just an extension, so do we want to keep it? Or has it been replaced by docs?

Comment (jpadams) [reply examples/nodejs/say.ts:1]: we'll rescue this code and move to `dagger/examples`.

**Pull request merged**